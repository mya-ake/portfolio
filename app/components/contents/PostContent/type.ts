export namespace PostContentType {
  export type Props = {
    slug: string;
  };
}
