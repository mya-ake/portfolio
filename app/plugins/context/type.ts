import { IHttp } from '@/lib/http';

export type Context = {
  http: IHttp<undefined>;
};
