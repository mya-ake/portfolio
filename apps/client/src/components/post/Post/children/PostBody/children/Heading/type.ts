import type { ReactNode } from 'react';

export type HeadingProps = {
  children: ReactNode;
};
