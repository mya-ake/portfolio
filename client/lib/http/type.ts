import { HTTP_METHODS } from './constants';

export type HttpMethods = typeof HTTP_METHODS[number];

export interface IHttp<E> {
  request<D>(
    request: HttpRequest,
    option?: HttpRequestOption,
  ): Promise<HttpResponse<D, E>>;

  cancelAll(): void;
  cancel(url: string): void;
}

export type HttpRequest = {
  url: string;
  method: HttpMethods;
};

export type HttpRequestOption = {
  cancelable: boolean;
};

interface IHttpResponse<D> {
  ok: boolean;
  status: number | undefined;
  data: D;
  noResponse: boolean;
  canceled: boolean;
}

export interface HttpOkResponse<D> extends IHttpResponse<D> {
  ok: true;
  status: number;
  noResponse: false;
  canceled: false;
}

export interface HttpErrorResponse<D> extends IHttpResponse<D> {
  ok: false;
  status: number;
  noResponse: false;
  canceled: false;
}

export interface HttpNoResponse extends IHttpResponse<undefined> {
  ok: false;
  status: undefined;
  data: undefined;
  noResponse: true;
  canceled: false;
}

export interface HttpCancelResponse extends IHttpResponse<undefined> {
  ok: false;
  status: undefined;
  data: undefined;
  noResponse: true;
  canceled: true;
}

export type HttpResponse<D, E> =
  | HttpOkResponse<D>
  | HttpErrorResponse<E>
  | HttpNoResponse
  | HttpCancelResponse;
