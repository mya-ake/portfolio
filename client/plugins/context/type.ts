import { IHttp } from '@/lib/http';

export type AppHttp = IHttp<undefined>;

export type AppContext = {
  http: AppHttp;
};
