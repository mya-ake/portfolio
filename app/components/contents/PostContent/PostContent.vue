<template>
  <div class="post-content">
    <template v-if="post">
      <div class="content" v-html="post.body"></div>
    </template>

    <template v-else-if="requesting">
      <p>Requesting...</p>
    </template>

    <template v-else>
      <p>No content</p>
    </template>
  </div>
</template>

<script lang="ts">
import { createComponent, onMounted } from 'vue-function-api';
import { usePost } from '@/hooks';
import { PostContentType } from './type';

export default createComponent({
  props: {
    slug: {},
  } as PostContentType.PropsType,

  setup({ slug }, { root }) {
    const { request, requesting, post, hasPost } = usePost(root.$_context);

    onMounted(async () => {
      await request(slug);
    });

    return {
      requesting,
      post,
      hasPost,
    };
  },
});
</script>

<style lang="scss" scoped>
.post-content {
  max-width: var(--width-content);
  margin: 0 auto;
  padding: 0 space(4);
  @include media(small) {
    padding: 0;
  }
}
</style>

<style lang="scss">
.post-content {
  p {
    margin: 1em 0;
  }
}
</style>
