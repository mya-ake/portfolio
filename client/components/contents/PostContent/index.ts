export { default as PostContent } from './PostContent.vue';
export * from './type';
