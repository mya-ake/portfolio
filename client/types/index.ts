export * from '@@/types/content.type';
export * from '@/plugins/context';
