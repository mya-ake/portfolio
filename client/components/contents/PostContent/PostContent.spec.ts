import { shallowMount, Wrapper } from '@vue/test-utils';
import PostContent from './PostContent.vue';
import { Content } from '@@/types/content.type';
import {
  createComponentLocalVue,
  createContextMock,
  createOkResponse,
  createPostData,
  clearMocks,
} from '@@/tests/fixtures';
import { CombinedVueInstance } from 'vue/types/vue';

const localVue = createComponentLocalVue();
const { mocks, createMock } = createContextMock();

afterEach(() => {
  clearMocks(mocks);
});

describe('components/contents/PostContent', () => {
  describe('mountable', () => {
    let wrapper: Wrapper<
      CombinedVueInstance<
        PostContent,
        object,
        object,
        object,
        Record<never, any>
      >
    >;

    beforeEach(() => {
      mocks.http.mockReturnValueOnce(
        createOkResponse<Content>({ data: createPostData() }),
      );

      wrapper = shallowMount(PostContent, {
        propsData: {
          slug: 'post-slug',
        },
        mocks: createMock(),
        localVue,
      });
    });

    it('is vue instance', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
