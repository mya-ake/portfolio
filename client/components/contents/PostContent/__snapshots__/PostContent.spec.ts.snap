// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/contents/PostContent mountable snapshot 1`] = `
<div>
  <div>
    <h1>
      post title
    </h1>
  </div>
</div>
`;
